### PR TITLE
Angular Folder Structure

### PR DESCRIPTION
**Pull request by Hinton**
Based on #6.

Noteworthy changes:

- Rewrote introduction.
- Added background resources for context
- Introduced `SharedModule` and `CoreModule`
- Updated the folder structure slightly to not have the initial `modules` folder

### CODE REVIEW

File: decisions/0011-angular-folder-structure.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+directory or something similar. However at this stage the number of directories we have is still
+maintainable and nesting things one step further would further indent the editor navigator. Thanks
+to the use of barrel files and angular modules such a change can be easily adopted at a later stage.
+
+This structure will be implemented in multiple steps.
+
+1. Extract unrelated files from `src/app`. https://github.com/bitwarden/clients/pull/3127
+2. Create `CoreModule`. https://github.com/bitwarden/clients/pull/3149
+3. Create `SharedModule`.
+4. Migrate all existing loose components to `SharedModule`.
```
Comment (MGibson1) [inline decisions/0011-angular-folder-structure.md:83]: Seems like this `SharedModule` would be the same as `LooseComponentsModule` is today. Is that accurate?

Comment (Hinton) [reply decisions/0011-angular-folder-structure.md:83]: In essence. The way they differ is that `SharedModule` will always have some components that needs to be shared. `LooseComponentsModule` is a tech debt module where we need to extract components until that module can be removed. Initially I'm aiming to retain `LooseComponentsModule` and have `SharedModule` import it.

**Comment by addisonbeck** [comment]
I think this looks great and manages to encapsulate all the things that each person wants to achieve the most from this refactor. The only criticism I really I have is that the example still makes use of a lot of `index.ts` files, even when there is an NgModule sibling next to it. I'm still not sure what we are really gaining from this? Importing modules has always worked out fine in my experience.

File: decisions/0011-angular-folder-structure.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  app.component.html
+  app.component.ts
+  app.module
+  oss-routing.module.ts
+  oss.module.ts
+  wildcard-routing.module.ts
+```
+
+An argument could be made that the root feature modules should be placed under a `features`
+directory or something similar. However at this stage the number of directories we have is still
```
Comment (addisonbeck) [inline decisions/0011-angular-folder-structure.md]: I mostly wanted to use a `features` or `modules` folder to separate components from global classes like singleton services and shared models. Using `shared` and `core` folders achieves the same thing. I think we can just remove this line from the ADR, the example structure will likely work for us for a long time.

**Comment by Hinton** [comment]
@addisonbeck barrel files still serve a purpose in my opinion. By combining them with https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-internal-modules.md we can prevent the usage of private code. Angular Modules to my knowledge doesn't provide that benefit, as any module can import any file.

**Review by MGibson1 — DISMISSED** [review]

**Review by MGibson1 — APPROVED** [review]

**Pull request merged**